JustSaying/JustSayingBus.cs

```diff
@@ -158,33 +161,42 @@ private IMessagePublisher GetActivePublisherForMessage(Message message)
                 throw new InvalidOperationException(errorMessage);
             }
 
-            string activeRegion;
-            if (Config.GetActiveRegion == null)
+            var activeRegion = GetActiveRegion();
+            if (!string.Equals(_previousRegion, activeRegion, StringComparison.Ordinal))
             {
-                activeRegion = Config.Regions.First();
+                _log.LogInformation("Active region for publishing has been set to '{Region}', was '{PreviousRegion}'.",
+                    activeRegion, _previousRegion);
+                _previousRegion = activeRegion;
             }
-            else
-            {
-                activeRegion = Config.GetActiveRegion();
-            }
-            _log.LogInformation("Active region has been evaluated to '{Region}'.", activeRegion);
 
-            if (!_publishersByRegionAndTopic.ContainsKey(activeRegion))
+            var publishersForRegionFound = _publishersByRegionAndTopic.TryGetValue(activeRegion, out var publishersForRegion);
+            if (!publishersForRegionFound)
             {
                 _log.LogError("Error publishing message. No publishers registered for active region '{Region}'.", activeRegion);
                 throw new InvalidOperationException($"Error publishing message. No publishers registered for active region '{activeRegion}'.");
             }
 
             var topic = message.GetType().ToTopicName();
-            var publishersByTopic = _publishersByRegionAndTopic[activeRegion];
-            if (!publishersByTopic.ContainsKey(topic))
+            var publisherFound = publishersForRegion.TryGetValue(topic, out var publisher);
+
+            if (!publisherFound)
             {
                 _log.LogError("Error publishing message. No publishers registered for message type '{MessageType}' in active region '{Region}'.",
                     message.GetType(), activeRegion);
                 throw new InvalidOperationException($"Error publishing message, no publishers registered for message type '{message.GetType()}' in active region '{activeRegion}'.");
             }
 
-            return publishersByTopic[topic];
+            return publisher;
+        }
+
+        private string GetActiveRegion()
+        {
+            if (Config.GetActiveRegion == null)
+            {
+                return Config.Regions.First();
```

Comment:
Is there code elsewhere that _guarantees_ this _always_ has at least one item in it? Otherwise the exception thrown would be very vague and generic.

Reply:
Yes, it's checked in `Validate.Validate`.